Добавляем функционал + тесты в конфигурацию

ProgrammSM opened the pull request:
Добавляем unit-тесты GridFormTest.php с проверкой загрузки конфигурации, добавления записей, удаления записей, редактирования записей, восстановления базы данных из резервной копии.
Добавляем резервные копии базы данных для восстановления.
Добавляем модель GridForm.php с методами getConfiguration(), addService(), removeServices(), editService(), restoreConfig().
Вносим изменения в ConfigController.php переписав actionGrid(), дополнив actionAdd(), actionRemove(), actionEdit(), actionRestore().
Вносим изменения в grid.php добавляя кнопки и javascript для формирования запросов.
Добавляем в params.php 'dbPath' содержащий абсолютный путь до директории с базой данных.
Добавляем файл test_params.php для изменения загрузки параметров в тестах.
Вносим изменения в test.php загружая $params из test_params.php.

Commit:
Добавляем функционал + тесты в конфигурацию

Добавляем unit-тесты GridFormTest.php с проверкой загрузки конфигурации, добавления записей, удаления записей, редактирования записей, восстановления базы данных из резервной копии.
Добавляем резервные копии базы данных для восстановления.
Добавляем модель GridForm.php с методами getConfiguration(), addService(), removeServices(), editService(), restoreConfig().
Вносим изменения в ConfigController.php переписав actionGrid(), дополнив actionAdd(), actionRemove(), actionEdit(), actionRestore().
Вносим изменения в grid.php добавляя кнопки и javascript для формирования запросов.
Добавляем в params.php 'dbPath' содержащий абсолютный путь до директории с базой данных.
Добавляем файл test_params.php для изменения загрузки параметров в тестах.
Вносим изменения в test.php загружая $params из test_params.php.

artezh commented on config/params.php:
```diff
@@ -1,6 +1,10 @@
 <?php
+$dbPath = realpath(__DIR__ . '/../data');
```
не надо в конфиге переменные делать

artezh commented on controllers/ConfigController.php:
```diff
@@ -14,12 +16,92 @@ class ConfigController extends Controller
+        $url = 'index.php?r=config%2Fgrid';
+        $model = new GridForm();
+        if (!$model->addService()) {
+            $url .= '&err=Ошибка%20добавления';
```
throw new HttpException(422, 'Ошибка добавления')

Commit:
Изменяем обработку ошибок, оптимизируем код.

Изменяем обработку ошибок в ConfigController.php.
Восстанавливаем actions() в SiteController.php для корректной работы обработчика ошибок.
Изменяем представление grid.php, выкидывая лишнее и исправляя ошибку в javascript.
Избавялемся от переменных в файлах конфигурации.

Commit:
Добавление потерянной строки после PHP-кода

artezh commented on controllers/ConfigController.php:
```diff
@@ -12,14 +15,96 @@
+     * @return \yii\web\Response
+     * @throws HttpException
+     */
+    public function actionRemove($id = null)
```
не надо предустановленное значение, если обязательный параметр

artezh commented on controllers/ConfigController.php:
```diff
@@ -12,14 +15,96 @@
+            if (!$model->removeServices(BaseJson::decode($id))) {
+                throw new HttpException(422, 'Ошибка удаления');
+            }
+        } else {
```
Соответственно второй эксепшн не нужен

artezh commented on models/GridForm.php:
```diff
@@ -0,0 +1,115 @@
+            // Задаём счётчик выполненных операций
+            $counter = 0;
+            foreach ($array as $id) {
+                $counter += \Yii::$app->db->createCommand()
```
Работу с базой переделать на модели

Commit:
Оптимизация кода

Переименована таблица services в service в базах данных и исправлены все обращения к ней.
В CalcForm.php заменяем обращение к базе данных.
В GridForm.php работа с БД исправляем на обращения через модель Service.
В Service.php добавляем сценарий для массового присваивания
В grid.php удаляем проверку количества выделенных чекбоксов при формировании json строки.
В ConfigController.php удаляем присваивание null в качестве значения по умолчанию у обязательных атрибутов action и удаляем некоторые исключения.

Commit:
Удаляем непонятно откуда взявшуюся константу

artezh commented on models/GridForm.php:
```diff
@@ -0,0 +1,99 @@
+     */
+    public function addService()
+    {
+        $service = new Service();
```
непонятно зачем

Commit:
Переписываем механизм добавления новой услуги

Переписываем тест GridFormTest.php с метода addService на editService.
В GridForm.php удаляем метод addService и переписываем editService под добавление новых услуг.
В grid.php изменяем идентификатор #strAdd на #srvAdd и переписываем его функционал на добавление новой строки-болванки в таблицу с возможностью дальнейшего сохранения.

Commit:
Изменяем механизм добавления услуг

Добавляем в GridFormTest.php тест testNewService(), testInvalidEditService(), testInvalidRemoveServices(), testEmptyEditService(), testEmptyRemoveServices(). Удаляем testAddService();
Добавляем в GridFrom.php метод newService($data), исправляем editService. В editService и removeServices добавляем проверку на существование id передаваемого пользователем и на пустоту входящего массива
В Service.php меняем название сценария
В grid.php переписываем javascript
В ConfigController.php добавляем actionNew($service)

Commit:
Исправляем комментарии и названия переменных

artezh commented on models/Service.php:
```diff
@@ -0,0 +1,24 @@
+ */
+class Service extends ActiveRecord
+{
+    const SCENARIO_EDIT = 'newOrEdit';
```
Два разных сценария

Commit:
Разделяем сценарии, правим комментарии, удаляем лишнее из шаблона

В Service.php разделяем сценарий на два для добавления и редактирования.
В Service.php, ConfigController.php, GridForm.php, SiteController.php, CalcForm.php.
Удалены модели и представления стандартного шаблона, которые не используются в проекте.
